### PR TITLE
Fix documentation typo

### PR DESCRIPTION
**Pull request by johnktims**

### CODE REVIEW

**Comment by johnktims** [comment]
I should have added `[ci skip]`.

**Comment by Shredder121** [comment]
Have you also looked at the Korean docs? /sarcasm

No it looks good, do you still want to weed through more of the documentation?

**Comment by timowest** [comment]
For this a backport variant could be created as well.

**Comment by Shredder121** [comment]
That is very true, and reminds me of [a question about the documentation](https://github.com/querydsl/querydsl/pull/1246#issuecomment-78156831).

**Comment by johnktims** [comment]
You can leave this one open for a little bit longer. I'm working on a different bug and have been fixing little typos as I find them. I'll backport this when it comes time to close it.

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java [hunk 1]
```diff
@@ -517,7 +517,7 @@ public void setBeanSuffix(String beanSuffix) {
     /**
      * Override the NamingStrategy (default: new DefaultNamingStrategy())
      *
-     * @param namingStrategy namingstrategy to override (default: new DefaultNamingStrategy())
+     * @param namingStrategy namingStrategy to override (default: new DefaultNamingStrategy())
```
Comment (johnktims) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java]: This should probably be separated by a space instead.

File: querydsl-core/src/main/java/com/querydsl/core/types/expr/Wildcard.java [hunk 2]
```diff
@@ -36,12 +36,12 @@
...
-     * Wilcard count distinct expression ((count(distinct *))
+     * Wildcard count distinct expression ((count(distinct *))
      */
     public static final NumberExpression<Long> countDistinct = NumberOperation.create(Long.class, Ops.AggOps.COUNT_DISTINCT_ALL_AGG);
 
     /**
-     * Wilcard count distinct expression ((count(distinct *))
+     * Wildcard count distinct expression ((count(distinct *))
      */
     public static final NumberExpression<Integer> countAsInt = NumberOperation.create(Integer.class, Ops.AggOps.COUNT_ALL_AGG);
```
Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/types/expr/Wildcard.java]: This is not count distinct :open_mouth: 

[see above](https://github.com/querydsl/querydsl/pull/1263/files#diff-daaa8ce4559ba899a54b3bc40fe9a8d9R34)

Comment (johnktims) [inline querydsl-core/src/main/java/com/querydsl/core/types/expr/Wildcard.java]: Good catch. I'll remove the references to `distinct`. Also the parentheses are unbalanced. Is that desired?

Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/types/expr/Wildcard.java]: Good catch as well. I think it is desirable to wrap it in parentheses, so adding one on the right is good.

**Comment by timowest** [comment]
@johnktims This looks good to me. Shall I merge it, or do you still want to push something?

File: querydsl-spatial/src/main/java/com/querydsl/spatial/path/GeometryPath.java [hunk 3]
```diff
@@ -139,7 +139,7 @@ public GeometryPath(String var) {
         return polygon;
     }
 
-    public PolyhedralSurfacePath<PolyHedralSurface> asPolygHedralSurface() {
+    public PolyhedralSurfacePath<PolyHedralSurface> asPolyHedralSurface() {
```
Comment (johnktims) [inline querydsl-spatial/src/main/java/com/querydsl/spatial/path/GeometryPath.java]: This should probably be `asPolyhedralSurface` since polyhedral is one word.

Comment (Shredder121) [inline querydsl-spatial/src/main/java/com/querydsl/spatial/path/GeometryPath.java]: If you were to fix this, I'd do it in a separate commit if I were you.
Otherwise you can't easily backport it.

Comment (johnktims) [inline querydsl-spatial/src/main/java/com/querydsl/spatial/path/GeometryPath.java]: I just noticed that Geolatte also uses PolyHedral so it might be best to leave it.

**Comment by johnktims** [comment]
I think this is good to go.

File: querydsl-docs/src/main/docbook/ko-KR/content/tutorials/scala.xml [hunk 4]
```diff
@@ -171,7 +171,7 @@ exporter.export(connection.getMetaData)
 query().from(employee).select(employee.firstName, employee.lastName)
 ]]></programlisting>
 
-      <para>다음과 같이 Employee 또는 QEmployee의 companio 객체를 사용할 수 있다.</para>
+      <para>다음과 같이 Employee 또는 QEmployee의 companion 객체를 사용할 수 있다.</para>
```
Comment (Shredder121) [inline querydsl-docs/src/main/docbook/ko-KR/content/tutorials/scala.xml]: You have nerves to search through the Korean docs like that haha.

File: querydsl-jpa/src/main/java/com/querydsl/jpa/Conversions.java [hunk 5]
```diff
@@ -86,10 +86,10 @@ private static boolean isEntityPathAndNeedsWrapping(Expression<?> expr) {
         } else if (Enum.class.isAssignableFrom(expr.getType())) {
             return new EnumConversion<RT>(expr);
         } else if (expr instanceof FactoryExpression) {
-            FactoryExpression<RT> factorye = (FactoryExpression<RT>)expr;
+            FactoryExpression<RT> factory = (FactoryExpression<RT>)expr;
             boolean numberConversions = false;
             boolean hasEntityPath = false;
-            for (Expression<?> e : factorye.getArgs()) {
+            for (Expression<?> e : factory.getArgs()) {
```
Comment (Shredder121) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/Conversions.java]: I'd suggest `factoryExpr` instead?

File: querydsl-docs/src/main/docbook/ko-KR/content/tutorials/scala.xml (hunk 4 above)
Comment (johnktims) [inline querydsl-docs/src/main/docbook/ko-KR/content/tutorials/scala.xml]: Haha. Since it references "employee" I though that maybe they meant "company" so I popped it into google translate and companion made the most sense. I could be totally wrong.

File: querydsl-spatial/src/main/java/com/querydsl/spatial/jts/JTSPolygonExpression.java [hunk 6]
```diff
@@ -40,7 +40,7 @@
     private volatile NumberExpression<Integer> numInteriorRing;
 
     @Nullable
-    private volatile JTSLineStringExpression<LineString> exterorRing;
+    private volatile JTSLineStringExpression<LineString> exteriorRing;
```
Comment (Shredder121) [inline querydsl-spatial/src/main/java/com/querydsl/spatial/jts/JTSPolygonExpression.java]: **This is not backportable.**

We rely on default serialization, renames will break the serialized form.
I suggest either while cherry-picking removing this(be sure to mention that this change was not included), or splitting the commits.

File: querydsl-docs/src/main/docbook/ko-KR/content/tutorials/scala.xml (hunk 4 above)
Comment (Shredder121) [inline querydsl-docs/src/main/docbook/ko-KR/content/tutorials/scala.xml]: No, it probably references the `QEmployee companion class` (query object).
But. I could also be wrong.

File: querydsl-spatial/src/main/java/com/querydsl/spatial/jts/JTSPolygonExpression.java (hunk 6 above)
Comment (johnktims) [inline querydsl-spatial/src/main/java/com/querydsl/spatial/jts/JTSPolygonExpression.java]: Good catch. So I should just leave a comment in the code not to change that misspelling when I backport this branch?

Comment (Shredder121) [inline querydsl-spatial/src/main/java/com/querydsl/spatial/jts/JTSPolygonExpression.java]: Yes, that and then remove it back again (so only a local change, no need to push comments)
or splitting the commit.

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java [hunk 7]
```diff
@@ -30,7 +30,7 @@
 import com.querydsl.sql.codegen.support.PrimaryKeyData;
 
 /**
- * ExtendedBeanSerialzier outputs primary key based equals, hashCode and toString implementations
+ * ExtendedBeanSerializer outputs primary key based equals, hashCode and toString implementations
```
Comment (Shredder121) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java]: Pro tip: if you use `{@link }` comments, IDEs should assist you in (not) making typos.
The same applies for [later on in the diff](https://github.com/querydsl/querydsl/pull/1263/files#diff-bb2c9aa8e1e2b3228ff0918b96f99dd0R35).

Comment (johnktims) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java]: Should I start updating those references in this pull request?

Comment (Shredder121) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java]: I think you could start with the current changed references yes.
@timowest what do you think? Should we change the current changed references, or in another PR? Or do you think it is not necessary?

Comment (timowest) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java]: Preferably another PR. Let's try to get some merges done also to make new work easier.

Comment (Shredder121) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java]: That is fine with me.

**Comment by johnktims** [comment]
Are there any changes that still need to be made here?

**Comment by Shredder121** [comment]
Do we have entries on the AntMetadataExporter in the reference documentation?
If there is a table with the configuration options, it is good to see if they match the Javadoc.

**Comment by timowest** [comment]
Wow, this covers quite a lot. It looks good to me. 

@Shredder121 The AndMetadataExporter config entries are the same as for the Maven plugin http://www.querydsl.com/static/querydsl/3.6.2/reference/html/ch02s03.html#d0e961

**Comment by Shredder121** [comment]
That's good, only one table to maintain.
Then I suppose the content of said table matches the Javadoc.
So in my opinion there are no further changes that need to be made here.

**Comment by timowest** [comment]
I put the reminder for the backport into the main comment.
